An Untitled Mining Game/An Untitled Mining Game/objects/obj_SaveTracker/KeyPress_75.gml

```diff
@@ -26,12 +26,21 @@ var i = 0; repeat(array_length(node_type_array)) {
 					save_entity.node_health = node_health;
 					save_entity.spr_to_draw = spr_to_draw;
 					break;
+			//-----Obj_items
+				case "obj_item":
+					save_entity.item_num = item_num;
+					save_entity.x_frame = x_frame;
+					save_entity.y_frame = y_frame;
+					break;
 			//-----Structs
 				case "obj_structure_mouse_tool":
 				case "obj_structure_pebble_refiner":
 					save_entity.structure_level = structure_level;
 					save_entity.in_use = in_use;
 					save_entity.structure_level = structure_level;
+					save_entity.wait_time = wait_time;
+					save_entity.current_step = current_step;
+					save_entity.menu_up = menu_up;
```

Comment:
Interesting to see how this switch statement is growing! Consider how nesting your switch statements may let you "factor out" some repeats in the future. As an example, if you can deduce that an object is a structure (the string contains "structure", or something), you may be able to encode all "common" attributes (wait time, step, etc) first, then go into a nested switch to do the specifics of each. This might reduce code repetition.

Reply:
Im not sure what you mean?

Reply:
Ahh, my explanation was definitely all over the place. Here is a little bit of psuedo-code that exemplifies what I intended to say. It shows how switch statements can be "nested"; I'm assuming this is allowed in GML, but there's every chance that it is disallowed.

```c
switch(myText){
   case "object_a": break;
   case "object_b": break;
   default:
       if("structure" in myText){
        //encode things common to all structures
            switch(myText){   // This switch will only get hit with structure-related strings
                case "my-structure-a":
                       //encoding specific to structure a
                       break;
                case 'my-structure-b':
                       //encoding specific to structure b
                       break;
            }
        }
}
```